.appveyor.yml

```diff
@@ -1,28 +1,49 @@
-os: Visual Studio 2015
-platform: x64
+skip_branch_with_pr: true
+image: Visual Studio 2017
+
+platform: 
+  - x64
+
 environment:
   global:
-    MINICONDA: "C:\\Miniconda37-x64"
+    PYTHON_ROOT: "C:\\Miniconda36-x64"
   matrix:
     - PYTHON_VERSION: 3.6
     - PYTHON_VERSION: 3.7
 
 install:
-  - set PATH=%MINICONDA%;%MINICONDA%\Scripts;C:\\msys64\usr\bin;%PATH%
+  # If there is a newer build queued for the same PR, cancel this one.
+  # The AppVeyor 'rollout builds' option is supposed to serve the same
+  # purpose but it is problematic because it tends to cancel builds pushed
+  # directly to master instead of just PR builds (or the converse).
+  # credits: JuliaLang developers.
+  - ps: if ($env:APPVEYOR_PULL_REQUEST_NUMBER -and $env:APPVEYOR_BUILD_NUMBER -ne ((Invoke-RestMethod `
+      https://ci.appveyor.com/api/projects/$env:APPVEYOR_ACCOUNT_NAME/$env:APPVEYOR_PROJECT_SLUG/history?recordsNumber=50).builds | `
+      Where-Object pullRequestId -eq $env:APPVEYOR_PULL_REQUEST_NUMBER)[0].buildNumber) { `
+        throw "There are newer queued builds for this pull request, failing early." }
+  
+  - SET PATH=%PYTHON_ROOT%;%PYTHON_ROOT%\Scripts;%PATH%
+  - CALL activate.bat base
```

Comment:
Can we remove the rollout thing? I don't think we need it.

Also, what does activate.bat do?

Reply:
rollout? You mean %PYTHON_ROOT%? That is how cmd on windows handles env variables. We need to add them first, so activate.bat is on the PATH.

activate.bat adds necessary custom paths for conda, so it will work properly.

Reply:
I guess this was a conda bug or a conda/appveyor interaction bug? Is there a specific issue that we can monitor?

Trying to simplify things as much as possible.

Reply:
https://github.com/conda/conda/issues/8865

https://github.com/conda/conda/issues/8836